CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/GalleryFragment.kt

```diff
@@ -102,54 +113,59 @@ class GalleryFragment internal constructor() : Fragment() {
 
         // Handle share button press
         view.findViewById<ImageButton>(R.id.share_button).setOnClickListener {
-            // Make sure that we have a file to share
-            mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
-
-                // Create a sharing intent
-                val intent = Intent().apply {
-                    // Infer media type from file extension
-                    val mediaType = MimeTypeMap.getSingleton()
-                            .getMimeTypeFromExtension(mediaFile.extension)
-                    // Get URI from our FileProvider implementation
-                    val uri = FileProvider.getUriForFile(
-                            view.context, BuildConfig.APPLICATION_ID + ".provider", mediaFile)
-                    // Set the appropriate intent extra, type, action and flags
-                    putExtra(Intent.EXTRA_STREAM, uri)
-                    type = mediaType
-                    action = Intent.ACTION_SEND
-                    flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
-                }
-
-                // Launch the intent letting the user choose which app to share with
-                startActivity(Intent.createChooser(intent, getString(R.string.share_hint)))
+
+            val mediaFile = mediaList[(mediaViewPager.currentItem)]
+
+            // Create a sharing intent
+            val intent = Intent().apply {
+                // Infer media type from file extension
+                val mediaType = MimeTypeMap.getSingleton()
+                        .getMimeTypeFromExtension(mediaFile.extension)
+                // Get URI from our FileProvider implementation
+                val uri = FileProvider.getUriForFile(
+                        view.context, BuildConfig.APPLICATION_ID + ".provider", mediaFile)
+                // Set the appropriate intent extra, type, action and flags
+                putExtra(Intent.EXTRA_STREAM, uri)
+                type = mediaType
+                action = Intent.ACTION_SEND
+                flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
             }
+
+            // Launch the intent letting the user choose which app to share with
+            startActivity(Intent.createChooser(intent, getString(R.string.share_hint)))
         }
 
         // Handle delete button press
         view.findViewById<ImageButton>(R.id.delete_button).setOnClickListener {
+
+            val mediaFile = mediaList[(mediaViewPager.currentItem)]
+
             AlertDialog.Builder(view.context, android.R.style.Theme_Material_Dialog)
                     .setTitle(getString(R.string.delete_title))
                     .setMessage(getString(R.string.delete_dialog))
                     .setIcon(android.R.drawable.ic_dialog_alert)
                     .setPositiveButton(android.R.string.yes) { _, _ ->
-                        mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
```

Comment:
I would keep this the way it was. You may know that the list is not empty, but you don't know what the state of the current item is.

Reply:
I don't think so it requires yet cause if the item is last and you're going to delete it then it'll come back to camera activity. Even I tested it from multiple scenarios and it's working fine. Could you please elaborate why it required.

Reply:
If we change how current item is chosen in the future (e.g. letting users open the gallery and go directly to a specific media item) then this will likely break, so this change introduces a maintenance burden that didn't exist before.

Reply:
ok well I'm changing it